qendpoint-core/src/main/java/com/the_qa_company/qendpoint/core/compact/bitmap/Bitmap375Big.java

```diff
@@ -460,11 +461,67 @@ public static long binarySearch(LongArray arr, long val) {
 		return min;
 	}
 
+	public static long binarySearchNew(LongArray arr, long val) {
+		long min = 0;
+		long max = arr.length();
+		long mid;
```

Comment:
We could introduce methods in LongArray to do the actual estimation there instead of here.

e.g.
```
long min = arr.lowerBound(val);
long max = arr.upperBound(val);
long mid = arr.estimatedLocation(val);
```

With default implementations at the interface level, or in some common super class.